zeppelin-web/src/app/notebook/paragraph/paragraph.controller.js

```diff
@@ -966,6 +966,117 @@ angular.module('zeppelinWebApp').controller('ParagraphCtrl', function($scope, $r
   };
 
   var setTable = function(data, refresh) {
+    function addButtonMenuEvent(button, menu) {
```

Comment:
Do you think we could separate this function and all the sub-functions of that feature inside its own service?